### PR TITLE
Eltwise->Transpose sinking for preprocessing ops

### PR DESCRIPTION
**Pull request by GlebKazantaev**
### Description
Added Eltwise->Transpose sinking transformation for cases when eltwise came from  preprocessing and it is safe to sink it. This sinking is needed by further optimizations Eltwise->Convolution optimizations.

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp [hunk 1]
```diff
@@ -61,6 +62,54 @@ std::shared_ptr<ngraph::opset6::Constant> get_reversed_order_constant(const std:
...
+        auto eltwise = pattern_to_output.at(eltwise_p).get_node_shared_ptr();
+        auto eltwise_const_input = pattern_to_output.at(eltwise_const_input_p);
+        auto eltwise_data_input = pattern_to_output.at(eltwise_data_input_p);
+        auto transpose = pattern_to_output.at(transpose_p).get_node_shared_ptr();
+
+        const auto & order_size = transpose->get_input_shape(1).at(0);
+        const auto & shape = eltwise_const_input.get_shape();
+        if (shape.size() != order_size && ov::shape_size(shape) != 1) {
+            // TODO: temporary restrictions
+            return false;
```
Comment (jane-intel) [inline src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp:89]: basically we can make an unsqueeze like we pull Reshape through FQ

File: src/tests/functional/inference_engine/transformations/preprocessing_fusion_tests.cpp [hunk 2]
```diff
@@ -672,3 +694,90 @@ TEST_F(TransformationTestsF, RICFusionSplitConcatDetectionNegative3) {
...
+
+        manager.register_pass<pass::MOCTransformations>(false);
+    }
+
+    {
+        auto input = create_param({ 1, 64, 64, 3 });
+        auto order = ngraph::opset6::Constant::create(ngraph::element::i64, ngraph::Shape{ 4 }, { 0, 3, 1, 2 });
+        auto transpose = std::make_shared<ngraph::opset6::Transpose>(input, order);
+        auto conv = create_conv(transpose, {6, 3, 3, 3});
+        function_ref = std::make_shared<Function>(NodeVector{ conv }, ParameterVector{ input });
```
Comment (jane-intel) [inline src/tests/functional/inference_engine/transformations/preprocessing_fusion_tests.cpp:778]: how accuracy is being checked? where did scale go?

File: src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp (hunk 1 above)
Comment (GlebKazantaev) [reply src/common/transformations/src/transformations/common_optimizations/transpose_sinking.cpp:89]: Yes, the implementation is limited but it is enough to move preprocessing eltwises through transpose.
Also I was thinking that this sinking will be later replaced by TF FE transpose sinking which supports BinaryElementwiseArithmetic operations.

File: src/tests/functional/inference_engine/transformations/preprocessing_fusion_tests.cpp (hunk 2 above)
Comment (GlebKazantaev) [reply src/tests/functional/inference_engine/transformations/preprocessing_fusion_tests.cpp:778]: `enable_accuracy_check` enabled accuracy comparison between original and processed functions. See line below.

**Review by jane-intel — APPROVED** [review]

**Pull request merged**